### PR TITLE
Refactor document preprocessor and enhance logging for debug

### PR DESCRIPTION
**Pull request by dusvyat**
Refactored the document preprocessor to improve clarity and modularity by separating methods for splitting, metadata handling, and ID generation.

Introduced logging in the knowledge base controller to enhance traceability of preprocessing configuration and query execution, which aids in debugging by providing detailed runtime insights. 
 
Updated unit tests to validate the new structure and logging logic, ensuring consistent behavior and proper metadata handling across different preprocessing scenarios.

Improved handling in map reduce summarizer, especially handling of edge cases such as None metadata
 

## Description

Please include a summary of the change and the issue it solves. 

Fixes #issue_number

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/interfaces/knowledge_base/preprocessing/document_preprocessor.py [hunk 1]
```diff
@@ -134,22 +140,20 @@ def _generate_context(self, chunk_content: str, full_document: str) -> str:
...
 
     def process_documents(self, documents: List[Document]) -> List[ProcessedChunk]:
         processed_chunks = []
 
         for doc in documents:
+            # Get content_column from metadata if available
+            content_column = doc.metadata.get('content_column') if doc.metadata else None
+
+            # Ensure document has an ID
+            if doc.id is None:
```
Comment (dusvyat) [inline mindsdb/interfaces/knowledge_base/preprocessing/document_preprocessor.py:154]: I noticed i the traces in CW we had null for id on chunks

File: mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py [hunk 2]
```diff
@@ -94,13 +94,16 @@ def output_keys(self) -> List[str]:
     def _get_document_ids_from_chunks(self, chunks: List[Document]) -> List[str]:
         unique_document_ids = set()
         document_ids = []
+        logger.debug(f"Processing {len(chunks)} chunks to extract document IDs")
         for chunk in chunks:
-            metadata = chunk.metadata
+            metadata = chunk.metadata or {}
```
Comment (dusvyat) [inline mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py:99]: in case metadata is coming through as None.

**Review by fshabashev — APPROVED** [review]

File: mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py [hunk 3]
```diff
@@ -154,11 +178,17 @@ def _call(
...
         for summary in summaries:
             source_id_to_summary[summary.source_id] = summary.content
+        logger.debug(f"Generated {len(source_id_to_summary)} summaries")
 
         # Update context chunks with document summaries.
         for chunk in context_chunks:
-            doc_id = chunk.metadata.get(self.doc_id_key, '')
-            summary = source_id_to_summary[str(doc_id)]
-            chunk.metadata['summary'] = summary
+            metadata = chunk.metadata or {}
```
Comment (fshabashev) [inline mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py:185]: This line looks a bit weird, is this a standard way to check for None?

Comment (dusvyat) [reply mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py]: modified it to be more explicit

File: mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py [hunk 4]
```diff
@@ -94,13 +94,19 @@ def output_keys(self) -> List[str]:
     def _get_document_ids_from_chunks(self, chunks: List[Document]) -> List[str]:
         unique_document_ids = set()
         document_ids = []
+        logger.debug(f"Processing {len(chunks)} chunks to extract document IDs")
         for chunk in chunks:
+            if not chunk.metadata:
+                chunk.metadata = {}
```
Comment (dusvyat) [inline mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py:100]: explicitly make sure metadata is not None

**Pull request merged**